pkg/compact/compact.go

```diff
@@ -879,10 +879,13 @@ func NewBucketCompactor(
 }
 
 // Compact runs compaction over bucket.
-func (c *BucketCompactor) Compact(ctx context.Context) error {
+func (c *BucketCompactor) Compact(ctx context.Context) (err error) {
 	defer func() {
-		if err := os.RemoveAll(c.compactDir); err != nil {
-			level.Error(c.logger).Log("msg", "failed to remove compaction work directory", "path", c.compactDir, "err", err)
+		if IsHaltError(err) {
```

Comment:
hmm actually something here is a little funny.. the `err` variable is never actually used or set, so this is always be false, no?